packages/backend/src/rpc/dispatcher.spec.ts

```diff
@@ -313,6 +337,94 @@ describe('RPCDispatcher', () => {
           );
         })
         .then(() => {
+          expect(dummyHandler).not.toHaveBeenCalled();
+          expect(specificHandler).toHaveBeenCalledTimes(1);
+          expect(app.getSingleton(ServiceContextEvents).disposeContext).toHaveBeenCalled();
+        });
+    });
+
+    it('should respond with error response from the last registered error handler when the RPC call fails with error that the handler can handle', () => {
+      let app = new App();
+
+      // prepare request / response
+      app.overrideService(
+        RequestInfo,
+        class MockRequestInfo extends RequestInfo {
+          req = mockRequest('test.method', {});
+          res = mockResponse();
+        },
+      );
+
+      // mock middleware call and prepare erroneous response
+      app.overrideSingleton(
```

Comment:
Can this be extracted to private method, as I see it duplicated 7 times across the tests?
```typescript
      // mock middleware call and prepare erroneous response
      app.overrideSingleton(
        RPCMiddlewareContainer,
        class MockRPCMiddlewareContainer extends RPCMiddlewareContainer {
          call = jest.fn(() => Promise.reject({ isJoi: true, details: 'Error details' }));
        },
      );
```

Reply:
IMO, we shouldn't extract it cause there are 5-6 different override cases in the test and it's not worth it to do it just for the different mocks. So, would stick with it for now.